src/mender-update/standalone/standalone.cpp

```diff
@@ -201,618 +222,435 @@ error::Error RemoveStateData(database::KeyValueDatabase &db) {
-		return {Result::FailedNothingDone, exp_header.error()};
-	}
-	auto &header = exp_header.value();
-
-	if (options != InstallOptions::NoStdout) {
-		cout << "Installing artifact..." << endl;
-	}
+	return error::NoError;
+}
 
-	if (header.header.payload_type == "") {
-		auto data = StateDataFromPayloadHeaderView(header);
-		return DoEmptyPayloadArtifact(main_context, data, options);
-	}
+error::Error StateMachine::SetStopAfterState(const string &state) {
+	using tf = common::state_machine::TransitionFlag;
+	using se = StateEvent;
+	auto &s = state_machine_;
 
-	update_module::UpdateModule update_module(main_context, header.header.payload_type);
+	// Replace transition in state machine in order to exit at given point.
+	if (state == "Download") {
+		s.AddTransition(save_artifact_install_state_, se::Success, exit_state_, tf::Immediate);
 
-	err = update_module.CleanAndPrepareFileTree(update_module.GetUpdateModuleWorkDir(), header);
-	if (err != error::NoError) {
-		err = err.FollowedBy(update_module.Cleanup());
-		return {Result::FailedNothingDone, err};
-	}
+	} else if (state == "ArtifactInstall") {
+		s.AddTransition(save_artifact_commit_state_, se::Success, exit_state_, tf::Immediate);
 
-	StateData data = StateDataFromPayloadHeaderView(header);
+	} else if (state == "ArtifactCommit") {
+		s.AddTransition(save_cleanup_state_, se::Success, exit_state_, tf::Immediate);
 
-	auto exp_matches = main_context.MatchesArtifactDepends(header.header);
-	if (!exp_matches) {
-		log::Error(exp_matches.error().String());
-		return {Result::FailedNothingDone, err};
-	} else if (!exp_matches.value()) {
-		// reasons already logged
-		return {Result::FailedNothingDone, err};
+	} else if (state != "") {
+		return context::MakeError(context::ValueError, "Cannot stop after unknown state " + state);
 	}
 
-	err = SaveStateData(main_context.GetMenderStoreDB(), data);
-	if (err != error::NoError) {
-		err = err.FollowedBy(update_module.Cleanup());
-		return {Result::FailedNothingDone, err};
-	}
+	return error::NoError;
+}
 
-	return DoInstallStates(main_context, data, parser, update_module);
+void StateMachine::StartOnRollback() {
+	start_state_ = &rollback_query_state_;
 }
 
-ResultAndError Commit(context::MenderContext &main_context) {
-	auto exp_in_progress = LoadStateData(main_context.GetMenderStoreDB());
-	if (!exp_in_progress) {
-		return {Result::FailedNothingDone, exp_in_progress.error()};
-	}
-	auto &in_progress = exp_in_progress.value();
+error::Error PrepareContext(Context &ctx) {
+	const auto &default_paths {ctx.main_context.GetConfig().paths};
+	ctx.script_runner.reset(new executor::ScriptRunner(
+		ctx.loop,
+		chrono::seconds {ctx.main_context.GetConfig().state_script_timeout_seconds},
+		chrono::seconds {ctx.main_context.GetConfig().state_script_retry_interval_seconds},
+		chrono::seconds {ctx.main_context.GetConfig().state_script_retry_timeout_seconds},
+		default_paths.GetArtScriptsPath(),
+		default_paths.GetRootfsScriptsPath()));
```

Comment:
Shouldn't these be user defined from the configuration?

Reply:
Indeed, good catch!

Reply:
Lol! I actually proceeded to fix it, but then I realized that `default_paths` comes from the conf, it's just confusingly named. Changing.